src/transformers/activations_tf.py

```diff
@@ -15,7 +15,20 @@
 import math
 
 import tensorflow as tf
-from packaging import version
+from packaging.version import parse
+
+
+try:
+    import tf_keras as keras
+except (ModuleNotFoundError, ImportError):
+    import keras
+
+    if parse(keras.__version__).major > 2:
+        raise ValueError(
+            "Your currently installed version of Keras is Keras 3, but this is not yet supported in "
+            "Transformers. Please install the backwards-compatible tf-keras package with "
+            "`pip install tf-keras`."
+        )
```

Comment:
I'm slightly concerned that raising this exception at the top-level of a module is going to cause lots of issues. Users, who might not even be using tensorflow with transformers, but will have it in their environment will start having exception raised if they do e.g. `from transformers import *` (or something less stupid which ends up loading this module).

Reply:
Yeah, I was worried about this too! I **think** that specific problem shouldn't happen, because if TF isn't available then all the TF objects will be dummies and TF-specific files shouldn't be executed. 

I'm still a bit unsure about the import from `modeling_tf_utils`, though - it feels dubious to me, but it was the only way I could think of to avoid pasting boilerplate everywhere. Maybe for peace of mind it'd be better to just do that, though, which would resolve this issue (and most of the other comments here)

Reply:
We can just install the different possibilities: no TF, old keras, new keras and check to see if the exceptions / warnings are egregious. I suspect the issue will not be users that don't have tensorflow, but for people that have it installed in their environment but aren't necessarily using it with transformers. 

As TF is always a pain to find the set of compatible libraries, I can imagine a lot of people complaining. 

Could we have the `if parse` logic put into a function and then add that as validation when people instantiate objects or use functions? 

Or perhaps, create something like `require_keras_lt_2` we can use to decorate objects?

Reply:
(Also in general I think the library maintainers segregated TF stuff pretty hard in the past to make sure that it didn't spam the console or allocate GPU memory)